src/ess/reduce/widgets/_switchable_widget.py

```diff
@@ -49,3 +50,17 @@ def value(self) -> Any:
     @value.setter
     def value(self, value: Any) -> None:
         self.wrapped.value = value
+
+    def set_fields(self, new_values: dict[str, Any]) -> None:
+        # Retrieve and set the enabled flag first
+        new_values = dict(new_values)
+        enabled_flag = new_values.pop('enabled', self.enabled)
+        if not isinstance(enabled_flag, bool):
+            raise ValueError(f"`enabled` must be a boolean, got {enabled_flag}")
+        self.enabled = enabled_flag
+        # Set the rest of the fields
+        set_fields(self.wrapped, new_values)
```

Comment:
Should we only fill the rest of the fields if the widget is enabled, or do we fill in anyway?

Reply:
We should restore the state even if they are disabled. So that user can easily switch it on and see the restored status.